lib/internal/modules/cjs/loader.js

```diff
@@ -394,7 +394,19 @@ function findLongestRegisteredExtension(filename) {
   return '.js';
 }
 
-function trySelf(parentPath, request) {
+function trySelf(parent, request) {
+  if (!parent) return false;
+
+  let parentPath;
+  if (parent.filename) {
+    parentPath = parent.filename;
+  } else if (parent.id === '<repl>' || parent.id === 'internal/preload') {
+    try {
+      parentPath = process.cwd() + path.sep;
+    } catch {}
```

Comment:
this is likely just my confusion; when would `process.cwd()` throw?

Reply:
It would throw in case of `ENOENT` errors.

Beofre wrapping  `process.cwd()` in this `try/catch` the `test/parallel/test-cwd-enoent-preload.js` was failing with error:

```bash
=== release test-cwd-enoent-preload ===                                       
Path: parallel/test-cwd-enoent-preload
internal/bootstrap/switches/does_own_process_state.js:129
    cachedCwd = rawMethods.cwd();
                           ^

Error: ENOENT: no such file or directory, uv_cwd
    at process.wrappedCwd [as cwd] (internal/bootstrap/switches/does_own_process_state.js:129:28)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1012:45)
    at Function.Module._load (internal/modules/cjs/loader.js:908:27)
    at Module.require (internal/modules/cjs/loader.js:1099:19)
    at Module._preloadModules (internal/modules/cjs/loader.js:1355:12)
    at loadPreloadModules (internal/bootstrap/pre_execution.js:448:5)
    at prepareMainThreadExecution (internal/bootstrap/pre_execution.js:71:3)
    at internal/main/eval_string.js:14:1 {
  errno: -2,
  code: 'ENOENT',
  syscall: 'uv_cwd'
}
assert.js:102
  throw new AssertionError(obj);
  ^

```

So if `process.cwd()` fails with `ENOENT` we return `MODULE_NOT_FOUND` and not `ENOENT`.

Reply:
Thanks, makes sense! Still not clear how that would be possible, but I appreciate the direction :-)

Reply:
Maybe I should add a comment for future reference